src/Auth0Client.ts

```diff
@@ -826,6 +826,10 @@ export default class Auth0Client {
   public async getTokenSilently(
     options: GetTokenSilentlyOptions = {}
   ): Promise<string | GetTokenSilentlyVerboseResponse> {
+    // Strip any undefined values.
+    Object.keys(options).forEach(key => options[key] === undefined && delete options[key])
```

Comment:
I believe this would be a breaking change.

Previously, passing any custom property with a value when constructing `Auth0Client`, and passing that same property but with undefined to `getTokenSilently`, would ensure the global value isn't used.

With this change, I expect the actual global value, passed when constructing `Auth0Client` to be sent instead.

Even tho I can see how this can be expected in some situations, I can also see how it could introduce potential issues on unexpected behavior.

To help you better understand this, this is what I am referring to:

```ts
const client = new Auth0Client({
  // typical config here
  myCustomParam: '123'
});
```

Important bits in the codebase to highlight here:

- https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L301: This will pull `myCustomParam` from the `options` and stores it as `this.customOptions`.

Now imagine the following code to retrieve a token:
```ts
const token = await client.getTokenSilently({
  // typical config here
  myCustomParam: undefined
});
```

Important bits in the codebase to highlight here:
- https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L1108: This will pull `myCustomParam` (which is set to undefined) from the options.

Next thing that's important here is this line https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L1113-L1114, here we are merging both objects together, in this example being:

```ts
{
  ... {myCustomParam: '123'},
  ... {myCustomParam: undefined}
}
```
The result here is
```ts
{
  myCustomParam: undefined,
}
```
However, with the changes introduced here, the outcome is 
```ts
{
  myCustomParam: '123',
}
```
This is because `myCustomParams was removed from the `options`, so it also wont be part of the `customOptions`

Reply:
I agree and was about to respond similarly; I think this would be breaking behaviour for people who rely on this behaviour, as @fullstackfool points out in the PR. Therefore I don't think we can accommodate this PR.

While there might be potentially something we could do later if we get the opportunity to rework this behaviour, the workaround here would be to strip out undesired `undefined` values before `getTokenSilently` is called.

Reply:
I expected as much as I dug into the api more but figured it would be useful to submit the PR for records nonetheless. 

Thank you for taking the time to consider and give feedback - hopefully can be addressed in the next major release.

Reply:
@fullstackfool Thank you for willing to help improve here!